.github/workflows/build.yaml

```diff
@@ -244,6 +244,33 @@ jobs:
         with:
           files: lcov.info
 
+  build-no-alloc:
+    runs-on: ubuntu-20.04
+    needs:
+      - "rustfmt"
+      - "markdown-lint"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/sgxsdk
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          override: true
+      - run: rustup component add rust-src --toolchain nightly-x86_64-unknown-linux-gnu
+      - uses: r7kamura/rust-problem-matchers@v1
+      - name: Build types with no alloc crate
+        # Some notes on this build command:
+        # - The vendored headers are used to get the necessary DCAP headers
+        # - The installed `tlibc` is used to get a compilable `time.h` for the target.
+        # - In the unlikely event that `thumbv7m-none-eabi` was installed with rustup, this would error out with
+        #   duplicate core symbols due to `-Z build-std=core`.
+        run: |
+          cargo metadata --no-deps --format-version=1 |  \
+            jq -r '.packages[].name' | \
+            grep -e types | \
+            xargs -n1 sh -c 'CFLAGS="-isystem${GITHUB_WORKSPACE}/core/build/headers -isystem/opt/intel/sgxsdk/include/tlibc" cargo build -Z build-std=core --target thumbv7m-none-eabi -p $0 || exit 255'
```

Comment:
How do you feel about vendoring the tlibc headers rather than depending on `/opt/intel/sgxsdk/include/tlibc`?

Reply:
vendored the tlibc headers, but still used the cflags.

I tried being smart about sending the tlibc if the SGX_SDK isn't found, but this got a bit nasty around the test enclave.
I'm fine if we want to rethink how the test enclave gets those paths, but implictly adding tlibc when SGX_SDK not found also felt like an odd inconsistent behavior.